src/api/app/models/bs_request.rb

```diff
@@ -32,12 +32,15 @@ class SaveError < APIException
   scope :to_accept, -> { where(state: 'new').where('accept_at < ?', DateTime.now) }
   # Scopes for collections
   scope :with_actions, -> { joins(:bs_request_actions).distinct.order(priority: :asc, id: :desc) }
-  scope :with_involved_projects, ->(project_ids) { where(bs_request_actions: { target_project_id: project_ids } ) }
-  scope :with_involved_packages, ->(package_ids) { where(bs_request_actions: { target_package_id: package_ids } ) }
+  scope :with_involved_projects, ->(project_ids) { where(bs_request_actions: { target_project_id: project_ids }) }
+  scope :with_involved_packages, ->(package_ids) { where(bs_request_actions: { target_package_id: package_ids }) }
+
+  scope :with_source_subprojects, ->(project_name) { where('bs_request_actions.source_project like ?', project_name) }
+  scope :with_target_subprojects, ->(project_name) { where('bs_request_actions.target_project like ?', project_name) }
 
   scope :in_states, ->(states) { where(state: states) }
-  scope :with_types, ->(types) { where('bs_request_actions.type in (?)', types) }
-  scope :from_source_project, ->(source_project) { where('bs_request_actions.source_project = ?', source_project) }
+  scope :with_types, ->(types) { joins(:bs_request_actions).where(bs_request_actions: { type: types }).distinct.order(priority: :asc, id: :desc) }
```

Comment:
Can you please use an ``includes`` combined with a ``references``

```
includes(:bs_request_actions).references(:bs_request_actions).where(bs_request_actions: { type: types }).distinct.order(priority: :asc, id: :desc) }
```

Btw: Why is the distinct and ordering now necessary?